src/checks/checker.cc

```diff
@@ -161,27 +161,31 @@ void checker::reap() {
 
       // Service check result.
       if (service_check == result.get_object_check_type()) {
+        service_map::iterator
+          it = service::services.find({result.get_hostname(),
+                                       result.get_service_description()});
+        if (it == service::services.end()) {
+          logger(log_runtime_warning, basic)
+            << "Warning: Check result queue contained results for "
+            << "service '" << result.get_service_description() << "' on "
+            << "host '" << result.get_hostname() << "', but the service "
+            << "could not be found! Perhaps you forgot to define the "
+            << "service in your config files ?";
+        }
         try {
           // Check if the service exists.
-          unsigned int host_id(get_host_id(result.get_hostname()));
-          unsigned int service_id(get_service_id(
-                                    result.get_hostname(),
-                                    result.get_service_description()));
-          service& svc(find_service(host_id, service_id));
-          // Process the check result.
           logger(dbg_checks, more)
             << "Handling check result for service '"
             << result.get_service_description() << "' on host '"
             << result.get_hostname() << "'...";
-          svc.handle_async_check_result(&result);
+          it->second->handle_async_check_result(&result);
         }
-        catch (std::exception const& e) {
+        catch (std::exception const &e) {
           logger(log_runtime_warning, basic)
-            << "Warning: Check result queue contained results for "
-            << "service '" << result.get_service_description() << "' on "
-            << "host '" << result.get_hostname() << "', but the service "
-            << "could not be found! Perhaps you forgot to define the "
-            << "service in your config files ?";
+            << "Check result queue errors for "
```

Comment:
Little note here bacause the logger tells about a warning and the message tells about error and the message is not the same as before. Maybe this is normal.